src-ui.v3/src/Main.hs

```diff
@@ -137,28 +139,25 @@ bodyElement4 = do
       text ")"
 
     -- search box
-    sVal <- elAttr "div" ("class" =: "item search right clearfix") $ mdo
-              let searchAttr = ("class" =: "input-search") <> ("placeholder" =: "search...")
-                  sCfg = TextInputConfig "text" "" never (constDyn searchAttr)
-                  sVal0 = _textInput_value searchPkg
+    searchInputE <- elAttr "div" ("class" =: "item search right clearfix") $ do
               divClass "text" $ text "Package Search"
-              searchPkg <- textInput sCfg
-              pure sVal0
+              sVal0 <- inputElement $ def & inputElementConfig_elementConfig . elementConfig_initialAttributes .~ fold
+                ["class" =: "input-search","placeholder" =: "search..."]
+              debounce 0.1 $ _inputElement_input sVal0
             
-    let sDynPkg = ffor2 sVal dynPackages0 (V.filter . packageContained) -- Dynamic t (Vector pkgN)
-        sDynMap = Map.fromList . (fmap (\p -> (pkgNToText p,p))) . V.toList <$> sDynPkg
-        dynAttrBool = demuxed (demux sVal) T.empty 
-    _ <- dyn $ sVal >>= (\val -> 
-           case "" == val of
-            True -> constDyn $ elAttr "ul" ("display" =: "none") $ text ""
-            False -> 
-              constDyn $ elDynAttr "ul" (fmap (\t -> if t then "display" =: "none" else "display" =: "list-item") dynAttrBool) $ do
-                           _ <- listWithKey sDynMap $ \pId _ -> do -- m (Dynamic (Map k a))
-                                 el "li" $ elAttr "a" ("href" =: ("#/package/" <> pId)) $ text pId
-                           pure ()
-  
-      )
-    
+    let
+      textToJSString = JSS.pack . T.unpack
+      jsStringToText = T.pack . JSS.unpack
```

Comment:
We should kill the above to functions and use the conversions from Data.JSString.Text instead. I didn't know about those. That's my fault! :)